### PR TITLE
(kubernetes) delete deployment

### PR DESCRIPTION
**Pull request by lwander**
@duftler PTAL

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy [hunk 1]
```diff
@@ -441,4 +442,22 @@ class KubernetesApiAdaptor {
...
+    def labels = deployment.spec.selector.matchLabels
+    return getPods(namespace, labels)
+  }
+
+  List<Pod> getPodsForReplicaSet(String namespace, ReplicaSet replicaSet) {
+    def labels = replicaSet.spec.selector.matchLabels
+    return getPods(namespace, labels)
+  }
+
+  String deploymentRevision(ReplicaSet replicaSet) {
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy]: Being picky here, but the order of the overloaded methods is:
```
getPodsForDeployment(..., Deployment)
getPodsForDeployment(..., ReplicaSet)
deploymentRevision(ReplicaSet)
deploymentRevision(Deployment)
```

Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy]: Should this be `getDeploymentRevision()`?

Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy]: I'll flip them

Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy]: Good point

**Pull request merged**